MapboxNavigation/NavigationMapView.swift

```diff
@@ -577,12 +576,19 @@ open class NavigationMapView: MGLMapView {
                 return shape(describingCasing: route, legIndex: legIndex)
             }
             
-            let coordsForLeg = coordinates[previousLegCongestionIndex..<previousLegCongestionIndex + legCongestion.count + 1]
-            let destination = coordinates.suffix(from: previousLegCongestionIndex + 1)
-            let segment = zip(coordsForLeg, destination).map { [$0.0, $0.1] }
-            let congestionSegments = Array(zip(segment, legCongestion))
+            // The last coord of the preceding step, is shared with the first coord of the next step.
+            // We don't need both.
+            var legCoordinates = Array(leg.steps.flatMap {
+                $0.coordinates?.suffix(from: 1)
+            }.joined())
+            
+            // We need to add the first coord of the route in.
+            if let firstCoord = leg.steps.first?.coordinates?.first {
+                legCoordinates.insert(firstCoord, at: 0)
+            }
             
-            previousLegCongestionIndex = legCongestion.count
+            let segments = zip(legCoordinates, legCoordinates.suffix(from: 1)).map { [$0.0, $0.1] }
+            let congestionSegments = Array(zip(segments, legCongestion))
```

Comment:
Can I bother you to add https://github.com/mapbox/mapbox-navigation-ios/pull/807#pullrequestreview-74974713 as comments?